packages/ui/src/components/va-chip/VaChip.vue

```diff
@@ -35,139 +35,136 @@
-import { RouterLinkMixin } from '../../mixins/RouterLinkMixin/RouterLinkMixin'
-import { StatefulMixin } from '../../mixins/StatefulMixin/StatefulMixin'
+import { useRouterLink, useRouterLinkProps } from '../../composables/useRouterLink'
 import useKeyboardOnlyFocus from '../../composables/useKeyboardOnlyFocus'
+import { useColors, useColorProps } from '../../composables/useColor'
+import { useStateful, useStatefulEmits, useStatefulProps } from '../../composables/useStateful'
 import VaIcon from '../va-icon'
 
-class ChipProps {
-  modelValue = prop<boolean>({ type: Boolean, default: true })
-  closeable = prop<boolean>({ type: Boolean, default: false })
-  color = prop<string>({ type: String, default: 'primary' })
-  outline = prop<boolean>({ type: Boolean, default: false })
-  flat = prop<boolean>({ type: Boolean, default: false })
-  icon = prop<string>({ type: String, default: '' })
-  disabled = prop<boolean>({ type: Boolean, default: false })
-  square = prop<boolean>({ type: Boolean, default: false })
-  tag = prop<string>({ type: String, default: 'span' })
-  size = prop<string>({
-    type: String,
-    default: 'medium',
-    validator: (value: string) => {
-      return ['medium', 'small', 'large'].includes(value)
-    },
-  })
-  shadow = prop<boolean>({ type: Boolean, default: false })
-}
-
-const ChipPropsMixin = Vue.with(ChipProps)
-
-@Options({
+export default defineComponent({
   name: 'VaChip',
+
   components: { VaIcon },
-  emits: ['focus'],
-})
-export default class VaChip extends mixins(
-  RouterLinkMixin,
-  StatefulMixin,
-  ColorMixin,
-  ChipPropsMixin,
-) {
-  SetupContext = setup(() => {
+
+  emits: [...useStatefulEmits, 'focus'],
+
+  props: {
+    ...useRouterLinkProps,
+    ...useColorProps,
+    ...useStatefulProps,
+    modelValue: { type: Boolean as PropType<boolean>, default: true },
+    closeable: { type: Boolean as PropType<boolean>, default: false },
+    outline: { type: Boolean as PropType<boolean>, default: false },
+    disabled: { type: Boolean as PropType<boolean>, default: false },
+    square: { type: Boolean as PropType<boolean>, default: false },
+    shadow: { type: Boolean as PropType<boolean>, default: false },
+    flat: { type: Boolean as PropType<boolean>, default: false },
+    icon: { type: String as PropType<string>, default: '' },
+    tag: { type: String as PropType<string>, default: 'span' },
+
+    size: {
+      type: String as PropType<string>,
+      default: 'medium',
+      validator: (value: string) => {
+        return ['small', 'medium', 'large'].includes(value)
+      },
+    },
+  },
+
+  setup (props, { emit }) {
     const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
 
-    return {
-      hasKeyboardFocus,
-      keyboardFocusListeners,
-    }
-  })
+    const { getColor } = useColors()
+    const colorComputed = computed(() => getColor(props.color))
+    const borderColor = computed(() => props.outline ? colorComputed : '')
```

Comment:
```suggestion
    const borderColor = computed(() => props.outline ? colorComputed.value : '')
```

Reply:
Thanks, overlooked this. Related to the question with reactive.